File: app/src/main/java/com/manadigital/recyclerview1/fragments/ExploreFragment.java
package com.manadigital.recyclerview1.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.manadigital.recyclerview1.R;
import com.manadigital.recyclerview1.adapters.ExploreAdapter;
import com.manadigital.recyclerview1.entities.Destination;

import java.util.ArrayList;
import java.util.List;

public class ExploreFragment extends Fragment {

    private RecyclerView recyclerView;
    private ExploreAdapter adapter;
    private List<Destination> destinations;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_explore, container, false);

        recyclerView = view.findViewById(R.id.recyclerViewTrendingDestinations);

        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        setupRecyclerView();
        setToolbarTitle();
    }

    private void setupRecyclerView() {
        destinations = createDestinations(); // Define tus destinos aquí
        adapter = new ExploreAdapter(destinations);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.HORIZONTAL, false));
        recyclerView.setAdapter(adapter);
    }

    private List<Destination> createDestinations() {
        // Define tus destinos aquí
        List<Destination> destinations = new ArrayList<>();
        destinations.add(new Destination(R.drawable.d_boracay, "Destino 1", "País 1"));
        destinations.add(new Destination(R.drawable.d_baros, "Destino 2", "País 2"));
        destinations.add(new Destination(R.drawable.d_bali, "Destino 3", "País 3"));
        destinations.add(new Destination(R.drawable.d_palawan, "Destino 4", "País 4"));
        return destinations;
    }

    private void setToolbarTitle() {
        // Get reference to the toolbar in MainActivity
        Toolbar toolbar = getActivity().findViewById(R.id.toolbar);

        // Get references to elements inside the toolbar
        TextView toolbarTitle = toolbar.findViewById(R.id.toolbar_title);
        ImageView puntitos = toolbar.findViewById(R.id.more_vert);

        // Set visibility as needed
        toolbarTitle.setText("Explore Destinations");
    }
}

File: app/src/main/java/com/manadigital/recyclerview1/fragments/ExploreFragment.kt
package com.manadigital.recyclerview1.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.manadigital.recyclerview1.R
import com.manadigital.recyclerview1.adapters.ExploreAdapter
import com.manadigital.recyclerview1.adapters.OfferSimpleAdapter
import com.manadigital.recyclerview1.entities.Destination
import com.manadigital.recyclerview1.entities.OfferSimple

class ExploreFragment : Fragment() {
    private lateinit var viewRoot: View
    private lateinit var trendingDestinationsRecyclerView: RecyclerView
    private lateinit var offersRecyclerView: RecyclerView
    private lateinit var destinations: List<Destination>
    private lateinit var offers: List<OfferSimple>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        viewRoot = inflater.inflate(R.layout.fragment_explore, container, false)
        return viewRoot
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // Setup Trending Destinations RecyclerView
        trendingDestinationsRecyclerView = viewRoot.findViewById(R.id.recyclerViewTrendingDestinations)
        destinations = listOf(
            Destination(R.drawable.d_boracay, "Boracay", "Philippines", R.drawable.diasboracay),
            Destination(R.drawable.d_baros, "Baros", "Maldives", R.drawable.diasboracay),
            Destination(R.drawable.d_bali, "Bali", "Indonesia", R.drawable.diasboracay),
            Destination(R.drawable.d_palawan, "Palawan", "Philippines", R.drawable.diasboracay)
        )
        val trendingDestinationsAdapter = ExploreAdapter(destinations)
        trendingDestinationsRecyclerView.adapter = trendingDestinationsAdapter
        trendingDestinationsRecyclerView.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)

        // Setup Offers RecyclerView
        offersRecyclerView = viewRoot.findViewById(R.id.offers_horizontalRV)
        offers = listOf(
            OfferSimple("20% de descuento para usuarios de Mastercard", R.drawable.mastercard),
            OfferSimple("25% de descuento con tus tarjetas Visa", R.drawable.visa)
        )
        val offersAdapter = OfferSimpleAdapter(offers)
        offersRecyclerView.adapter = offersAdapter
        offersRecyclerView.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
    }
}
